restic/internal/command_executor.py

```diff
@@ -6,14 +6,19 @@
 logger = logging.getLogger(__name__)
 
 
-def execute(cmd):
+def execute(cmd, binary_mode=False):
     logger.debug('Executing restic command: %s', str(cmd))
     try:
-        process = subprocess.run(cmd,
-                                 capture_output=True,
-                                 text=True,
-                                 check=False,
-                                 encoding='utf-8')
+        subprocess_args = {
+            'capture_output': True,
+            'text': True,
+            'encoding': 'utf-8'
+        }
+        if binary_mode:
```

Comment:
`binary_mode` complicates the logic a lot for both this file and the unit test code. Why do we need it? Are there any alternatives?

Reply:
That's because `restic cat pack` returns binary data and cannot be parsed as text/utf-8. The only alternative I see is, to make all commands use the "binary mode". e2e tests passed, but I'm unclear about other implications of this change.